src/jupyterhub_config/custom_docker_spawner.py

```diff
@@ -229,24 +231,42 @@ def _configure_environment(self, user_dir: Path, user_env_dir: Path, username: s
 
         self.environment["SHELL"] = "/usr/bin/bash"
 
-        if self._is_rw_minio_user():
-            self.log.info(
-                f"MinIO read/write user detected: {self.user.name}. Setting up minio_rw credentials."
+        # Get user-specific MinIO credentials from governance service
+        # Check if KBASE_AUTH_TOKEN is available (should be set by pre_spawn_start)
+        kbase_token = self.environment.get("KBASE_AUTH_TOKEN")
+        if not kbase_token:
+            raise ValueError(
+                "KBASE_AUTH_TOKEN not found in spawner environment. Authentication required for governance client."
```

Comment:
Should this and the other credential fetching stuff be inside the if block below, so that this error occurs only if use_data_gov_creds is true?

Reply:
Ok, the error message is slightly misleading in that case, but not a huge deal as this shouldn't happen IIUC

Reply:
It's talking about the governance client makes it sound like user_data_gov_creds is on. Like I said, minor